### PR TITLE
ENH handle short sequences

### PR DESCRIPTION
**Pull request by celiosantosjr**
Macrel was trained using sequences with 10 characters, 
some times this sequence was composed by 8 residues plus 
start and stop. In this sense, the minimum length of a sequence
in residues is 8. Shorter peptides should be handled in a different
way.

Fix error in the features script that breaks when
sequences smaller than 8 residues are received.

-- Standard behavior is considering these small peptides as NAMPs.

### CODE REVIEW

File: macrel/AMP_features.py [hunk 1]
```diff
@@ -14,7 +14,14 @@
 
 GROUPS_SA = ['ALFCGIVW', 'RKQEND', 'MSPTHY'] #solventaccess
 GROUPS_HB = ['ILVWAMGT', 'FYSQCN', 'PHKEDR'] # HEIJNE&BLOMBERG1979
-
+colist = ["tinyAA", "smallAA", "aliphaticAA",
```
Comment (luispedro) [inline macrel/AMP_features.py:17]: I'm not sure I like this, tbh. It feels more error prone to move the definition further away from the use.

**Review by luispedro — CHANGES_REQUESTED** [review]
Can you (1) revert the bit about moving `colist` to outside of where it's used and (2) some documentation about this _minimum of 8 AA_ limitation?

Why 8 amino acids?

**Comment by luispedro** [comment]
Also, obviously, fix the issues with the CI

(You can use a personal fork to test the CI before opening the PR)

**Comment by celiosantosjr** [comment]
> Can you (1) revert the bit about moving `colist` to outside of where it's used and (2) some documentation about this _minimum of 8 AA_ limitation?
> 
> Why 8 amino acids?

It is 8 amino acids because the shorter sequence we can handle is 10 codons, therefore 10 excluding the start and stop, leaves 8.

File: macrel/AMP_features.py (hunk 1 above)
Comment (celiosantosjr) [reply macrel/AMP_features.py]: already took care of this, and put the list right above the loop using it.

**Comment by celiosantosjr** [comment]
## General

The commit messages can be simplified in terms of number. However, the main problem was that after fixing the issue of the short sequence. There was a pytest introduced inside macrel/macrel that was breaking the entire thing. Then, I substituted it with a test in the tests/ folder (short_seqs) where we test even a higher number of short sequences, covering all possible edge cases.

## Keynote:

*The current version has this modification to handle short sequences as non-amps automatically. Thus,  these short sequences can only be verified in the output if using the --keep-negatives flag.*

To predict peptides with a wider length range,  please check other systems, such as ampir.

**Review by celiosantosjr — COMMENTED** [review]
That was done, thanks.

**Comment by luispedro** [comment]
> The commit messages can be simplified in terms of number. 

Yes, please do. This should be a commit or two (commits should correspond to logical sets of changes).

> However, the main problem was that after fixing the issue of the short sequence. There was a pytest introduced inside macrel/macrel that was breaking the entire thing.

No, you broke the code, the test caught it, and the answer cannot be to remove the test. You changed the macrel interface and this would potentially break downstream uses. It would break things like the webapp &c.

This fix should **not** change anything that was previously working: anything that worked before should keep working exactly the same.

If there is a reason to update the interface, this needs to be done properly (a separate PR, updated docs, &c), not sneaked in in a supposed bugfix.

**Comment by luispedro** [comment]
Looking more at this, it seems that `v1.0` did handle peptides as short as 3 AAs. Maybe that was not done properly, but that is a change in behaviour, which I would prefer to keep somewhat separate from the fix to handle very short peptides

**Comment by celiosantosjr** [comment]
> Looking more at this, it seems that `v1.0` did handle peptides as short as 3 AAs. Maybe that was not done properly, but that is a change in behaviour, which I would prefer to keep somewhat separate from the fix to handle very short peptides

Leave the issue as it is then? Or just handle peptides 3 amino acids or less?

**Comment by luispedro** [comment]
My proposal for fixing this is https://github.com/BigDataBiology/macrel/pull/43

Just fixes the issue, does not change the behaviour for any other case
